folding/src/columns.rs

```diff
@@ -26,3 +25,53 @@ pub enum ExtendedFoldingColumn<C: FoldingConfig> {
     /// while folding/accumulating multiple expressions.
     Selector(C::Selector),
 }
+
+impl<C: FoldingConfig> std::hash::Hash for ExtendedFoldingColumn<C> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        core::mem::discriminant(self).hash(state);
```

Comment:
Only a comment: https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-fieldless-enumerations. Good reading.